### PR TITLE
fix(downtimes) Prevent the user from creating downtimes with start date, end date and duration after 2037

### PR DESCRIPTION
**Pull request by callapa**
## Description

Prevent the user from creating downtimes with start date, end date and duration after 2037.

Controls added on :
- UI interfaces
- API
- CLAPI

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 20.04.x
- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by jeremyjaouen — APPROVED** [review]

File: www/include/monitoring/downtime/AddDowntime.php [hunk 1]
```diff
@@ -543,6 +547,7 @@
          */
         $tpl = new Smarty();
         $tpl = initSmartyTpl($path, $tpl, "template/");
+        $tpl->assign('dataPickerMaxYear', DOWNTIME_YEAR_MAX - 1);
```
Comment (a-launois) [inline www/include/monitoring/downtime/AddDowntime.php:550]:
```suggestion
        $tpl->assign('datePickerMaxYear', DOWNTIME_YEAR_MAX - 1);
```

File: www/include/monitoring/downtime/template/AddDowntime.ihtml [hunk 2]
```diff
@@ -161,6 +161,9 @@
     });
     jQuery(".timepicker").timepicker();
     initDatepicker();
+{/literal}
+    jQuery('.datepicker').datepicker('option', 'maxDate', '12/31/{$dataPickerMaxYear}');
```
Comment (a-launois) [inline www/include/monitoring/downtime/template/AddDowntime.ihtml:165]:
```suggestion
    jQuery('.datepicker').datepicker('option', 'maxDate', '12/31/{$datePickerMaxYear}');
```

File: www/include/monitoring/external_cmd/popup/templates/massive_downtime.ihtml [hunk 3]
```diff
@@ -71,6 +71,9 @@
     });
     jQuery(".timepicker").timepicker();
     initDatepicker();
+    {/literal}
+        jQuery('.datepicker').datepicker('option', 'maxDate', '12/31/{$dataPickerMaxYear}');
```
Comment (a-launois) [inline www/include/monitoring/external_cmd/popup/templates/massive_downtime.ihtml:75]:
```suggestion
        jQuery('.datepicker').datepicker('option', 'maxDate', '12/31/{$datePickerMaxYear}');
```

File: www/include/monitoring/external_cmd/popup/massive_downtime.php [hunk 4]
```diff
@@ -281,4 +278,5 @@
 $tpl->assign('p', $p);
 $tpl->assign('cmd', $cmd);
 $tpl->assign('select', $select);
+$tpl->assign('dataPickerMaxYear', DOWNTIME_YEAR_MAX - 1);
```
Comment (a-launois) [inline www/include/monitoring/external_cmd/popup/massive_downtime.php:281]:
```suggestion
$tpl->assign('datePickerMaxYear', DOWNTIME_YEAR_MAX - 1);
```

**Review by adr-mo — APPROVED** [review]

File: www/include/monitoring/external_cmd/popup/massive_downtime.php (hunk 4 above)
Comment (adr-mo) [reply www/include/monitoring/external_cmd/popup/massive_downtime.php:281]: GG ! +1 did not see this one !

File: www/include/monitoring/external_cmd/popup/templates/massive_downtime.ihtml (hunk 3 above)
Comment (adr-mo) [reply www/include/monitoring/external_cmd/popup/templates/massive_downtime.ihtml:75]: GG ! +1 did not see this one !

File: www/include/monitoring/downtime/template/AddDowntime.ihtml (hunk 2 above)
Comment (adr-mo) [reply www/include/monitoring/downtime/template/AddDowntime.ihtml:165]: GG ! +1 did not see this one !

File: www/include/monitoring/downtime/AddDowntime.php (hunk 1 above)
Comment (adr-mo) [reply www/include/monitoring/downtime/AddDowntime.php:550]: GG ! +1 did not see this one !

File: src/Centreon/Application/Controller/DowntimeController.php [hunk 5]
```diff
@@ -705,4 +652,39 @@ private function hasDtRightsForResource(Contact $contact, ResourceEntity $resouc
...
     }
+
+    /**
+     * This function will ensure that the POST data is valid regarding validation constraints defined.
+     *
+     * @param Request $request
+     * @param string $jsonValidatorFile
+     * @throws \InvalidArgumentException
+     */
+    private function validateOrFail(Request $request, string $jsonValidatorFile): void
```
Comment (jeremyjaouen) [inline src/Centreon/Application/Controller/DowntimeController.php:663]: This could be directly in AbstractController

**Pull request merged**